### PR TITLE
MGMT 3854 - e2e test to log info

### PR DESCRIPTION
**Pull request by slaviered**
/hold
/@assign @YuviGold

### CODE REVIEW

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 1]
```diff
@@ -239,6 +239,15 @@ def set_proxy_values(self, http_proxy, https_proxy='', no_proxy=''):
     def start_install(self):
         self.api_client.install_cluster(cluster_id=self.id)
 
+    def wait_for_logs_complete(self, timeout, interval=60, host_only=False):
```
Comment (osherdp) [inline discovery-infra/test_infra/helper_classes/cluster.py:242]: it's not clear what ``host_only`` means in this context

File: discovery-infra/test_infra/logs_utils.py [hunk 2]
```diff
@@ -32,18 +33,14 @@ def verify_logs_uploaded(cluster_tar_path, expected_min_log_num, installation_su
...
-
-    assert False, "oc logs were not uploaded"
+def wait_and_verify_oc_logs_uploaded(cluster, cluster_tar_path):
+    try:
+        cluster.wait_for_logs_complete(timeout=OC_DOWNLOAD_LOGS_TIMEOUT, interval=OC_DOWNLOAD_LOGS_INTERVAL, host_only=False)
+        cluster.download_installation_logs(cluster_tar_path)
+        assert os.path.exists(cluster_tar_path), f"{cluster_tar_path} doesn't exist"
+        _verify_oc_logs_uploaded(cluster_tar_path)
+    except BaseException as err:
+        assert False, f'oc logs were not uploaded {str(err)}'
```
Comment (osherdp) [inline discovery-infra/test_infra/logs_utils.py:43]: it's a bit problematic, to lose the traceback info this way

File: discovery-infra/test_infra/utils.py [hunk 3]
```diff
@@ -382,6 +382,48 @@ def are_host_progress_in_stage(hosts, stages, nodes_count=1):
         f"hosts statuses are {host_info}")
     return False
 
+def wait_for_logs_complete(
+        client, cluster_id, timeout, interval=60, host_only=False
+):
+    log.info("wait till logs of cluster % are collected (or timed-out)", cluster_id)
```
Comment (osherdp) [inline discovery-infra/test_infra/utils.py:388]: %s
?

File: discovery-infra/test_infra/utils.py [hunk 4]
```diff
@@ -382,6 +382,48 @@ def are_host_progress_in_stage(hosts, stages, nodes_count=1):
...
+            statuses=statuses,
+            host_only=host_only
+        ),
+        timeout_seconds=timeout,
+        sleep_seconds=interval,
+        waiting_for="Logs to be in status %s" % statuses,
+        )
+        log.info("logs are in expected state")
+    except BaseException:
+        log.error("waiting for logs expired after %d", timeout)
```
Comment (osherdp) [inline discovery-infra/test_infra/utils.py:404]: it's necessarily a timeout?
I think you should remove this

File: discovery-infra/test_infra/utils.py [hunk 5]
```diff
@@ -382,6 +382,48 @@ def are_host_progress_in_stage(hosts, stages, nodes_count=1):
...
+    try:
+        cluster = client.cluster_get(cluster_id)
+        hosts = client.get_cluster_hosts(cluster_id)
+        cluster_logs_status = cluster.logs_info
+        host_logs_statuses = [host.get("logs_info", "") for host in hosts]
+        if all(s in statuses for s in host_logs_statuses):
+            if host_only:
+                log.info(f"found expected statuses. host logs: {host_logs_statuses}")
+                return True
+            else:
```
Comment (osherdp) [inline discovery-infra/test_infra/utils.py:417]: delete

File: discovery-infra/test_infra/utils.py [hunk 6]
```diff
@@ -382,6 +382,48 @@ def are_host_progress_in_stage(hosts, stages, nodes_count=1):
...
+        log.error("waiting for logs expired after %d", timeout)
+        raise
+
+def are_logs_in_status(client, cluster_id, statuses, host_only=False):
+    try:
+        cluster = client.cluster_get(cluster_id)
+        hosts = client.get_cluster_hosts(cluster_id)
+        cluster_logs_status = cluster.logs_info
+        host_logs_statuses = [host.get("logs_info", "") for host in hosts]
+        if all(s in statuses for s in host_logs_statuses):
```
Comment (osherdp) [inline discovery-infra/test_infra/utils.py:413]: let's do the opposite
```
if any... / not all...:
    return False
    
# rest of the code, without the redundant indentation
```

File: discovery-infra/test_infra/utils.py [hunk 7]
```diff
@@ -382,6 +382,48 @@ def are_host_progress_in_stage(hosts, stages, nodes_count=1):
...
+                return True
+            else:
+                if cluster_logs_status in statuses:
+                    log.info(f"found expected state. cluster logs: {cluster_logs_status}, host logs: {host_logs_statuses}")
+                    return True
+
+        log.info(f"Cluster logs not yet in their required state. cluster logs: {cluster_logs_status}, host logs: {host_logs_statuses}")
+        return False
+    except:
+        log.exception("Failed to get cluster %s log info", cluster_id)
```
Comment (osherdp) [inline discovery-infra/test_infra/utils.py:425]: in this case we're returning None

Comment (osherdp) [inline discovery-infra/test_infra/utils.py:425]: why we're swallowing the exception here?

File: discovery-infra/test_infra/utils.py [hunk 8]
```diff
@@ -382,6 +382,48 @@ def are_host_progress_in_stage(hosts, stages, nodes_count=1):
...
+
+def are_logs_in_status(client, cluster_id, statuses, host_only=False):
+    try:
+        cluster = client.cluster_get(cluster_id)
+        hosts = client.get_cluster_hosts(cluster_id)
+        cluster_logs_status = cluster.logs_info
+        host_logs_statuses = [host.get("logs_info", "") for host in hosts]
+        if all(s in statuses for s in host_logs_statuses):
+            if host_only:
+                log.info(f"found expected statuses. host logs: {host_logs_statuses}")
```
Comment (osherdp) [inline discovery-infra/test_infra/utils.py:415]: either use f-strings or %s, don't mix them in this file

**Review by osherdp — CHANGES_REQUESTED** [review]

File: discovery-infra/test_infra/helper_classes/cluster.py (hunk 1 above)
Comment (slaviered) [reply discovery-infra/test_infra/helper_classes/cluster.py:242]: just check the host logs to save waiting for timeout

File: discovery-infra/test_infra/logs_utils.py (hunk 2 above)
Comment (YuviGold) [inline discovery-infra/test_infra/logs_utils.py:43]: Should treat wrapping exceptions like [this](https://docs.python.org/3/library/exceptions.html#built-in-exceptions)

```suggestion
    except BaseException as err:
        raise ConnectionError("oc logs were not uploaded") from err
```

File: discovery-infra/test_infra/helper_classes/cluster.py (hunk 1 above)
Comment (slaviered) [reply discovery-infra/test_infra/helper_classes/cluster.py:242]: replaced to check_host_logs-only

File: discovery-infra/test_infra/logs_utils.py [hunk 9]
```diff
@@ -6,8 +6,9 @@
 
 from test_infra.consts import NUMBER_OF_MASTERS
 
-OC_DOWNLOAD_LOGS_INTERVAL = 10 * 60
-NUM_OF_RETRIES = 6
+OC_DOWNLOAD_LOGS_INTERVAL = 5 * 60
+OC_DOWNLOAD_LOGS_TIMEOUT = 60 * 60
```
Comment (slaviered) [inline discovery-infra/test_infra/logs_utils.py:10]: perhaps increase to 75 minutes

**Review by YuviGold — APPROVED** [review]
LGTM

**Pull request merged**